frontend/layouts/discover-page/helpers.ts

```diff
@@ -47,3 +58,123 @@ export const useSortingByOptions = (target?: SortingByTargetType): SortingOption
     ...(sortingOptions[target] || []),
   ]) as SortingOptionType[];
 };
+
+const pathFilters = {
+  [Paths.Projects]: [
+    'filter[category]',
+    'filter[sdg]',
+    'filter[instrument_type]',
+    'filter[ticket_size]',
+    'filter[impact]',
+    'filter[only_verified]',
+    'filter[full_text]',
+    'filter[priority_landscape]',
+  ],
+  [Paths.ProjectDevelopers]: [
+    'filter[category]',
+    'filter[impact]',
+    'filter[full_text]',
+    'filter[priority_landscape]',
+  ],
+  [Paths.Investors]: [
+    'filter[category]',
+    'filter[sdg]',
+    'filter[instrument_type]',
+    'filter[ticket_size]',
+    'filter[impact]',
+    'filter[full_text]',
+  ],
+  [Paths.OpenCalls]: [
+    'filter[sdg]',
+    'filter[instrument_type]',
+    'filter[ticket_size]',
+    'filter[only_verified]',
+    'filter[full_text]',
+  ],
+};
+
+const getPathParams = (
+  path: string,
+  pathname: string,
+  queryParams: { [key: string]: string | string[] }
+) => {
+  const { page, sorting, search, ...filters } = queryParams;
+
+  // Get only applicable filters for each path to avoid unnecessary fetches
+  const pathFilter = pickBy(filters, (_, key) => pathFilters[path].includes(key));
+
+  const params: { [key: string]: string | string[] } = { page, search, ...pathFilter };
+
+  if (pathname !== path) {
+    return {
+      ...params,
+      fields: ['name'],
+      perPage: 1,
+      'page[size]': 1,
+    };
+  }
+
+  // Use sorting value only when applies to path to avoid unnecessary fetches
+  if (path === Paths.Projects || sorting?.includes('name') || sorting?.includes('created_at')) {
+    params.sorting = sorting;
+  }
```

Comment:
I fail to see why this condition is needed. Can't we always send the `sorting` parameter?

When the user changes tab, the sorting options are reset, so we'll never send a sorting option that doesn't make sense for the current tab.

Reply:
The sorting options are reset when the tab changes, but when a tab's parameters change, like sorting, it triggers fetching all contents of tabs with the current params.

currently on staging: 
<img width="946" alt="Screenshot 2023-01-16 at 09 40 04" src="https://user-images.githubusercontent.com/48164343/212635732-0d2aee73-9c71-47b4-bbf2-b935aab8641e.png">